python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -109,7 +118,12 @@ cdef class DeviceScalar:
         device scalar.
         """
         cdef libcudf_types.data_type cdtype = self.get_raw_ptr()[0].type()
-        return cudf_to_np_types[<underlying_type_t_type_id>(cdtype.id())]
+        if cdtype.id() == libcudf_types.DECIMAL64:
+            return cudf.Decimal64Dtype(
+                cudf.Decimal64Dtype.MAX_PRECISION, -cdtype.scale()
+            )
```

Comment:
Does this mean that whatever dtype is passed into the constructor doesn't propagate throughout?

Reply:
Yeah that is a problem, good point, that we can at least mostly fix. When we construct from the host, we could make it so that we keep the precision of the dtype that was passed and propagate that. 

I think it will need to be `cudf.Decimal64Dtype.MAX_PRECISION` for scalars that are generated device side however, such as in the case of `get_element` since there's no precision in libcudf.

Reply:
In the case of `get_element` we should use the precision from the source column. The only case I can think of where we really won't know the precision is if there's something like a cast from integer --> decimal or string --> decimal done in libcudf and it infers the scale automatically.

Reply:
👍 working on a solution.

Reply:
I added a `dtype_metadata` dict to our `DeviceScalar` class and overrode `get_element` in `DecimalColumn` to copy into that metadata dict. Hopefully the dict will come in handy when constructing scalars from nested types.